### ISSUE TITLE
Support for reading Apache Aurora serverssets from Zookeeper

### ISSUE THREAD

**Issue by SEJeff** [EmileVauge/traefik#9]
[Aurora](http://aurora.apache.org/) is a metaframework for Mesos, much like Marathon. It writes service discovery to zookeeper in a standardized format twitter invented for writing to zookeeper named server sets.

It would be wonderful to use a tool such as this for users of Apache Aurora. Thanks!

**Comment by thorhs** [comment]
I would like this as well.

**Comment by EmileVauge** [comment]
Hi
Maybe It would be better to interact with Zookeeper directly instead no ?

**Comment by thorhs** [comment]
Do you mean the clients connecting to the service should do the service lookup through ZooKeeper?  If so, then that can be problematic since it pulls in ZooKeeper dependencies to all clients.  Many of my clients today are just plain cURL/bash scripts that would be difficult to interface with ZK.

**Comment by EmileVauge** [comment]
No, that's not what I wanted to say. Aurora does not provide any rest API to interact with. As Aurora puts its jobs in Zookeper, I was thinking of getting the conf directly from Zookeper :)

**Comment by SEJeff** [comment]
Right, Aurora will publish metadata for service discovery about every task it launches as a service in zookeeper. What we are asking for is that traefik be taught to read that metadata and do load balancing / proxying to the instances.

**Comment by EmileVauge** [comment]
OK, but I will need some help, or some time :)

**Comment by SEJeff** [comment]
This is the most pcommonly used library](https://github.com/samuel/go-zookeeper) for go + zk. I could write a small python script to create the "server set" znodes for you if that would be a good starting point?

**Comment by SEJeff** [comment]
#25 is relevant somewhat as `libkv` supports zookeeper.

**Comment by EmileVauge** [comment]
You're absolutely right :) But it will need some adaptations to support Apache Aurora standard. It will not run out of the box.

**Comment by SEJeff** [comment]
@EmileVauge Now the serverset data would be written by Aurora and should be read-only as far as traefik is concerned. Would it make more sense to just build support for it in natively?

**Comment by EmileVauge** [comment]
Hi, as Zookeeper is now supported, it should be site easy to get Aurora on the same way :) Help is welcomed :)
